src/Raven.Server/SqlMigration/SqlDatabase.Validator.cs

```diff
@@ -0,0 +1,198 @@
+                    }
+                    else count++;
+                }
+            }
+
+            private void ValidateEmbeddedTables(List<SqlTable> tables)
+            {
+                foreach (var table in tables)
+                {
+                    foreach (var embeddedTable in table.EmbeddedTables)
+                    {
+                        if (!embeddedTable.ForeignKeys.ContainsValue(table.Name) && tables.Contains(embeddedTable))
+                            AddError($"Table '{embeddedTable.Name}' cannot embed into '{table.Name}'");
+                    }
+                }
+            }
+
+            private void ValidateDuplicateTables(List<SqlTable> tables)
+            {
+                var names = new List<string>();
+
+                foreach (var table in tables)
+                {
+                    if (table.IsEmbedded)
+                        continue;
+
+                    if (names.Contains(table.Name))
+                        AddError($"Duplicate table '{table.Name}'");
+
+                    else 
+                        names.Add(table.Name);
+
+                    ValidateDuplicateEmbeddedTables(table.EmbeddedTables);
+                }
+            }
+
+            private void ValidateDuplicateEmbeddedTables(List<SqlEmbeddedTable> tableEmbeddedTables)
+            {
+                var properties = new List<string>();
+                var tables = new List<SqlTable>();
+
+                foreach (var embeddedTable in tableEmbeddedTables)
+                {
+                    if (properties.Contains(embeddedTable.Property))
+                        AddError($"Duplicate property name '{embeddedTable.Property}'");
+
+                    else
+                        properties.Add(embeddedTable.Property);
+
+                    if (tables.Contains(embeddedTable))
+                        AddError($"Duplicate table '{embeddedTable.Name}' (try give them property name)");
+
+                    else
+                        tables.Add(embeddedTable);
+
+                    ValidateDuplicateEmbeddedTables(embeddedTable.EmbeddedTables);
+                }
+            }
+
+            private void ValidatePatch(SqlTable table, RavenDocument document)
+            {
+                try
+                {
+                    table.GetJsPatcher()?.PatchDocument(document);
+                }
+                catch
+                {
+                    AddError($"Cannot patch table '{table.Name}' using the given script");
+                }
+            }
+
+            private void ValidateHasPrimaryKeys(SqlTable table)
+            {
+                if (table.PrimaryKeys.Count == 0)
+                    AddError($"Table '{table.Name}' must have at list 1 primary key");
+            }
+
+            private void ValidateQuery(SqlTable table, out RavenDocument document)
+            {
+                document = null;
+
+                if (table.InitialQuery.ToLower().Contains(" order by "))
```

Comment:
Use `IndexOf` with a `StringComparison`, instead

Reply:
This will also not work if the query contains things like `ORDER       BY` or `select 'Order BY' as value`